test/helpers/assertFailure.js

```diff
@@ -1,9 +1,13 @@
 const { expect } = require('chai');
+const AssertionError = require('assert').AssertionError;
 
 async function assertFailure (promise) {
   try {
     await promise;
   } catch (error) {
+    if (error instanceof AssertionError) {
+      expect(error.message).to.equal('Wrong failure type');
```

Comment:
I don't think it's right to hardcode the message string in this function. Could you add an optional message argument to `assertFailure`, and use it in place of the string here? If the argument
is undefined we should skip the check.

Reply:
@frangio i think it isn't right to hard code the message here, too.
but in future, maybe we throw another AssertionError with another message at another place, so pass the message as an argument to `assertFailure` then check may not the right solution here.
what we actually want is confirm the AssertionError was well format or not, that's all. is it ok if i keep `assertFailure` as original and add an test case like bellow?

```js
it('rejects a revert with incorrect expected reason', async function () {
      await expectRevert(this.reverter.revertFromRevertWithReason(), 'Another wrong reason')
        .then(assert.fail)
        .catch((error) => {
          expect(error).to.be.an.instanceof(assert.AssertionError);
          expect(error.message).to.equal('Wrong failure type');
          expect(error.expected).to.equal('Another wrong reason');
          expect(error.actual).to.include('revert');
        });
    });
```

Reply:
I completely agree that what we want is to confirm that `AssertionError` is well formatted. Adding only one new test sounds good to me! Though I think it would be nicer to reuse `assertFailure` by slightly modifying it to return the caught error, instead of manually `.then`/`.catch`ing.